CMakeLists.txt

```diff
@@ -0,0 +1,34 @@
+cmake_minimum_required(VERSION 3.10)
+project(usgscsm VERSION 0.0.1 DESCRIPTION "usgscsm library")
+
+include(GNUInstallDirs)
+add_library(usgscsm SHARED 
+            src/UsgsAstroFramePlugin.cpp
+            src/UsgsAstroFrameSensorModel.cpp
+            src/UsgsAstroLsPlugin.cpp
+            src/UsgsAstroLsSensorModel.cpp
+            src/UsgsAstroLsStateData.cpp)
+
+set_target_properties(usgscsm PROPERTIES
+    VERSION ${PROJECT_VERSION}
+    SOVERSION 1
+    PUBLIC_HEADER include/usgscsm/UsgsAstroFramePlugin.h
+    PUBLIC_HEADER include/usgscsm/UsgsAstroFrameSensorModel.h
+    PUBLIC_HEADER include/usgscsm/UsgsAstroLsPlugin.h
+    PUBLIC_HEADER include/usgscsm/UsgsAstroLsSensorModel.h
```

Comment:
I think we should either make all of our headers public to make testing easier, or make none of them public.